exporters/otlp/lib/opentelemetry/exporters/otlp/exporter.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk'
+require 'net/http'
+
+require 'opentelemetry/proto/common/v1/common_pb'
+require 'opentelemetry/proto/resource/v1/resource_pb'
+require 'opentelemetry/proto/trace/v1/trace_pb'
+require 'opentelemetry/proto/collector/trace/v1/trace_service_pb'
+
+module OpenTelemetry
+  module Exporters
+    module OTLP
+      # An OpenTelemetry trace exporter that sends spans over HTTP as Protobuf encoded OTLP ExportTraceServiceRequests.
+      class Exporter
+        SUCCESS = OpenTelemetry::SDK::Trace::Export::SUCCESS
+        FAILURE = OpenTelemetry::SDK::Trace::Export::FAILURE
+        private_constant(:SUCCESS, :FAILURE)
+
+        # Default timeouts in seconds.
+        KEEP_ALIVE_TIMEOUT = 30
+        OPEN_TIMEOUT = 5
+        READ_TIMEOUT = 5
+        RETRY_COUNT = 5
+        PATH = '/v1/trace'
+        private_constant(:KEEP_ALIVE_TIMEOUT, :OPEN_TIMEOUT, :READ_TIMEOUT, :RETRY_COUNT, :PATH)
+
+        def initialize(host:,
+                       port:,
```

Comment:
from a vendor perspective, we have found that there's a benefit to having host (and port, to a lesser extent) be something that can be configured via a standardised environment variable.

I am unfortunately _not_ the best person to explain the use cases here but one example where, i believe, it has value would be in k8s, if you have a collector running ...(_gestures vaguely_) in each node...per...cluster?

In that situation, the application container may not know the host ip a collector would be listening on ahead of time and it would need to be set as an env var using the downward api

- https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api

- https://docs.datadoghq.com/agent/kubernetes/apm/?tab=daemonset#setup
```
Configure your application pods to pull the host IP in order to communicate with the Datadog Agent: Use the downward API to pull the host IP; the application container needs the DD_AGENT_HOST environment variable that points to status.hostIP.

    apiVersion: apps/v1
    kind: Deployment
     # ...
        spec:
          containers:
          - name: "<CONTAINER_NAME>"
            image: "<CONTAINER_IMAGE>"/"<TAG>"
            env:
              - name: DD_AGENT_HOST
                valueFrom:
                  fieldRef:
                    fieldPath: status.hostIP
```

Just one example but I'm sure there's many others where env var's make life easier for users, or would be required to use in a production environment. 

Obviously, fetching an env var and passing in it's value when initialising the otlp exporter is something a user could roll themselves quite easily. But as things scale having something standard and that get's picked up by default is a tremendous help for reducing onboarding pain.

looking at https://github.com/open-telemetry/opentelemetry-collector/tree/master/examples/demo i see something called `OTEL_AGENT_ENDPOINT` in the docker-compose....is this a standard?

Anyway, if there's something in the spec around this i think it's worth adding support for it. Also, the value of `OTEL_AGENT_ENDPOINT` in that example implies the spec may end up supporting a `url` value that would need to have host/port extracted from it.

otoh, if there's nothing in the spec it may just be that this implementation right now is constrained by that, but if there's a reasonable or already somewhat adopted env var, adding that support in early i think will help users

Reply:
https://github.com/open-telemetry/opentelemetry-specification/pull/699 is an attempt to specify all the things for OTLP, with a particular focus on environment variables for configuration. When that lands, this should be a lot clearer.